test-tools/issuer-front-end/src/Main.tsx

```diff
@@ -90,7 +90,71 @@ async function addRevokationKey(
     }
 }
 
-type AttributeDetails = { tag: string; type: string; value: string | undefined };
+type AttributeDetails = { tag: string; type: string; value: string | undefined; required: boolean };
+
+function renderAddPrompt(details: AttributeDetails) {
+    if (details.required) {
+        return (
+            <div>
+                {' '}
+                Add <b className="text-warning">required</b> attribute <strong> {details.tag} </strong>{' '}
+            </div>
+        );
+    }
+    return (
+        <div>
+            {' '}
+            Add <strong> {details.tag} </strong>{' '}
+        </div>
+    );
+}
+
+async function extractFromSchema(url: string): Promise<AttributeDetails[]> {
+    const response = await fetch(EXAMPLE_CREDENTIAL_SCHEMA);
+    if (!response.ok) {
+        throw new Error(`Unable to get schema. Response code: ${response.status}`);
+    }
+    const json = await response.json();
+    const { properties, required } = json.properties.credentialSubject.properties.attributes;
+
+    const attributeSchemaValues: AttributeDetails[] = [];
+    Object.entries(properties).forEach(([key, obj]) => {
+        attributeSchemaValues.push({
+            tag: key,
+            type: (obj as { type: string }).type,
+            value: undefined,
+            required: (required as string[]).includes(key),
+        });
+    });
+    return attributeSchemaValues;
+}
+
+function parseAttributesFromForm(
+    attributeSchema: AttributeDetails[],
+    setParsingError: (msg: string) => void
+): Attribute {
+    const attributes: Attribute = {};
+    attributeSchema.forEach((obj) => {
+        if (obj.required && obj.value === undefined) {
+            setParsingError(`Attribute ${obj.tag} needs to be set.`);
+            throw new Error(`Attribute ${obj.tag} needs to be set.`);
+        } else if (obj.value !== undefined) {
+            if (obj.type === 'string') {
+                attributes[obj.tag] = obj.value;
+            } else if (obj.type === 'number') {
+                attributes[obj.tag] = BigInt(obj.value);
+            } else {
+                setParsingError(
+                    `Attribute ${obj.tag} has type ${obj.type}. Only the types string/number are supported.`
+                );
+                throw new Error(
+                    `Attribute ${obj.tag} has type ${obj.type}. Only the types string/number are supported.`
+                );
+            }
+        }
+    });
+    return attributes;
+}
```

Comment:
@DOBEN We should go with your PR since it also adds a bit more error handling.

But can you copy these helpers in there and use them to reduce the amount of code duplication?